keps/sig-architecture/2140-receipts/README.md

```diff
@@ -0,0 +1,464 @@
+
+#### API Reviewers
+
+* Review / Approve Exception Requests that need API Review
+    * Ensure reviewer bandwidth considered
+    * Prioritize areas for upcoming releases
+
+#### SIG Chair / TLs / PMs
+
+* Review Open/Proposed KEPs for my SIG
+    * Help plan reviews / dependencies
+    * Help identify major themes / release blogs
+* Review At Risk KEPS for my SIG
+    * Help prioritize last-minute reviews
+* Review and create Exceptions for my SIG
+    * Determine release-blocking / primary importance
+* Review Slipped Enhancements to support planning
+    * Determine what phase it missed
+    * Identify multiple slips / misses
+
+#### Enhancements Subteam
+
+* Pre enhancements Freeze (Enhancements Collection)
+  * Collect intentions/plans of KEP authors to graduate a KEP
+  * Ensure that those intents comply with all requirements of the KEP
+  * Mark enhancements "At Risk" based on compliance with all requirements of the KEP
+
+* At Enhancements Freeze
+  * Remove enhancements that do not comply with requirements
+
+* Enhancements Freeze - Code Freeze
+  * Ensure that the k/k PRs for the KEP are populated
+  * Mark enhancements "At Risk" based on the progress of implementation
+  * Approve/Deny exceptions for Enhancements Freeze
+
+* Around Code Freeze
+  * Ensure that the enlisted k/k are merged
+  * Approve/Deny exceptions for Code Freeze
+
+* At Code Freeze
+  * Remove enhancements that have not completed requirements for Code Freeze
+
+* Throughout the Cycle
+  * Prepare a manifest of what's the status of enhancements at any point in time
+
+#### Docs Subteam
+
+* Enhancements Freeze - Open Docs PR Deadline
+  * Ensure that the k/w PRs are populated in the KEPs
+  * Communicate to SIG Docs about the expected review load
+
+* Open Docs PR Deadline - Docs Complete
+  * Mark enhancements "At Risk" if the following are true:
+    *  Docs PRs are not updated
+    *  The draft PRs have not been updated
+
+### Notes/Constraints/Caveats
+
+This KEP is based on a [high-level plan](https://docs.google.com/document/d/1qnfXjQCBrikbbu9F38hdzWEo5ZDBd57Cqi0wG0V6aGc/edit#) incorporating feedback collected from release team members and contributors over several cycles.
+
+In 1.22, we'll focus on implementing much of this process manually, so there will be some burden placed on KEP Authors to create the appropriate YAML structure. But ideally, these changes will free up time for people on the release team to work on automating the receipts process.
+
+### Risks and Mitigations
+
+* Non-compliance to the process by stakeholders like SIGs/KEP Authors
+  * Stricter gates have to be put for the process
+  * Exceptions only approach for a receipt to be added back to release cycle stage after the deadline
+  * Enhancement Team + Release Leads should have the final say on exceptions. The decision needs to be taken after considering the point of view of respective SIG leadership.
+* Conflicts can arise when two people update/modify the same artifact
```

Comment:
Also conflicts between places similar meta-data is stored. We have this issue with prr-approvers in kep.yaml as well as in the PRR approval request. We are going to eliminate the one in the kep.yaml. You'll want to make sure there are not redundant places people have to put in the meta-data, so you may want to remove milestones, latest-milestone from kep.yaml

Reply:
That is a nice point.

The ideology that we were following is that kep.yaml is the source of truth for all KEP related information. The receipts are just snaphots in time of what happened to a KEP. So, we can eventually deprecate the milestones struct and latest-milestones from kep.yaml once Receipts is up and running.